File: (04-08-2024)Group Anagrams/joshlin/solution.ts
function groupAnagrams(strs: string[]): string[][] {
  const cache = new Map<string, string[]>();
  for (const word of strs) {
      const key = word.split('').sort().join('');
      if (!cache.has(key)) cache.set(key, [word]);
      else cache.get(key)?.push(word);
  };
  return Array.from(cache.values());
};

File: (04-08-2024)Group Anagrams/joshlin/solution.cpp
#include <unordered_map>
#include <vector>
#include <string>
#include <algorithm>

using namespace std;

class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> cache;
        vector<vector<string>> res;     
        for (const auto& word: strs) {
            string key = word;
            sort(key.begin(), key.end());
            cache[key].push_back(word);
        };
        for (const auto& [_, group]: cache) {
            res.push_back(group);
        };
        return res;
    };
};